src/Raven.Server/ServerWide/Commands/RegisterReplicationHubAccessCommand.cs

```diff
@@ -0,0 +1,125 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Raven.Client.Documents.Operations.Replication;
+using Raven.Client.Util;
+using Sparrow.Json;
+using Sparrow.Json.Parsing;
+
+namespace Raven.Server.ServerWide.Commands
+{
+    public class BulkRegisterReplicationHubAccessCommand : CommandBase
+    {
+        public List<RegisterReplicationHubAccessCommand> Commands;
+        public string Database;
+
+        public BulkRegisterReplicationHubAccessCommand() : base(RaftIdGenerator.DontCareId)
+        {
+            
+        }
+        
+        public override DynamicJsonValue ToJson(JsonOperationContext context)
+        {
+            var djv = base.ToJson(context);
+            djv[nameof(Database)] = Database;
+            djv[nameof(Commands)] = new DynamicJsonArray(Commands.Select(x=>x.ToJson(context)));
+            return djv;
+        }
+    }
+    
+    public class RegisterReplicationHubAccessCommand : CommandBase
+    {
+        public string Database;
+        public string HubDefinitionName;
+        public string CertPublicKeyHash;
+        public string CertThumbprint;
+        public string CertificateBase64;
+        public string[] Incoming;
+        public string[] Outgoing;
+        public string Name;
+        public DateTime NotBefore, NotAfter;
+        public string Issuer;
+        public string Subject;
+        public bool RegisteringSamePublicKeyPinningHash;
+
+        public RegisterReplicationHubAccessCommand()
+        {
+            // for deserialization
+        }
+
+        public RegisterReplicationHubAccessCommand(
+            string databaseName, 
+            string hub, 
+            ReplicationHubAccess access, 
+            string certPublicKeyHash, 
+            string certThumbprint,
+            string uniqueRequestId,
+            string issuer,
+            string subject,
+            DateTime notBefore,
+            DateTime notAfter)
+        {
+            UniqueRequestId = uniqueRequestId;
+            Database = databaseName;
+            HubDefinitionName= hub;
+            CertPublicKeyHash = certPublicKeyHash;
+            CertThumbprint = certThumbprint;
+            CertificateBase64 = access.CertificateBas64;
+            Name = access.Name;
+            Incoming = access.Incoming;
+            Outgoing = access.Outgoing;
+            NotBefore = notBefore;
+            NotAfter = notAfter;
+            Issuer = issuer;
+            Subject = subject;
+        }
+
+        public override DynamicJsonValue ToJson(JsonOperationContext context)
+        {
+            var djv = base.ToJson(context);
+            djv[nameof(Name)] = Name;
+            djv[nameof(HubDefinitionName)] = HubDefinitionName;
```

Comment:
it is HubDefinitionName or HubName actually?

Reply:
Yes, fixed

Reply:
It was not fixed?